Assets/Scripts/EnemyController.cs

```diff
@@ -106,15 +111,27 @@ public void OnDamageTaken()
         gameObject.GetComponent<Renderer>().material.color = Color.Lerp(lowHealthCol, maxHealthCol, health / maxHealth);
     }
 
-    IEnumerator TriggerDamageEffect()
+    IEnumerator doAttack()
     {
         //Time damage effect delay to when attack happens
         yield return new WaitForSeconds(attackAnimationDelay);
         if (distanceToPlayer <= minDist)
         {
             //player.GetComponent<HurtEffect>().Hit();
-            HitPlayer(player.gameObject, -attackDamage);
-            Debug.LogError("Player is attacked");
+            if (PhotonNetwork.IsMasterClient)
+            { 
+                HitPlayer(player.gameObject, -attackDamage);
+            }
+        }
+        else if(distanceToPlayer <= shootingDistance)   //if player too far, shoot instead
+        {
+            GameObject projectile;
+            projectile = Instantiate(projectilePrefab, transform.position, Quaternion.identity);
+            projectile.GetComponent<EnemyProjectile>().enemyWhoShot = this.gameObject;
+            projectile.GetComponent<EnemyProjectile>().damage = this.attackDamage;
+            projectile.GetComponent<EnemyProjectile>().target = player;
+            projectile.GetComponent<EnemyProjectile>().isLocal = PhotonNetwork.IsMasterClient;
+            projectile.GetComponent<EnemyProjectile>().Launch();
         }
 
```

Comment:
Do we have/need some logic of the player also being in a view for the enemy for him to shoot then? So if they're behind the wall, so they wouldn't start shooting

Reply:
We discussed, there is no point to have this